### PR TITLE
create roc picture and corrplot pictures and also trees for this assi…

### PR DESCRIPTION
**Pull request by edwardwangxy**
…gnment.

Not sure about if the calculation is correct or not

### CODE REVIEW

File: Assignment7.Rmd [hunk 1]
```diff
@@ -9,27 +9,48 @@ In the following assignment you will be looking at data from an one level of an
...
 
 #Visualization 
 ```{r}
 #Start by creating histograms of the distributions for all variables (#HINT: look up "facet" in the ggplot documentation)
-
+library(ggplot2)
+library(reshape2)
+library(corrplot)
+meltD = melt(D1)
+hisplot = ggplot(meltD,aes(x=value))
```
Comment (charles-lang) [inline Assignment7.Rmd]: Hey Edward, this looks good. Could you upload this image as well? I can't see it.

File: Assignment7.Rmd [hunk 2]
```diff
@@ -41,30 +62,52 @@ plot(performance(pred.detail, "tpr", "fpr"))
...
 
 ```
 #Thresholds
 ```{r}
 #Look at the ROC plot for your first model. Based on this plot choose a probability threshold that balances capturing the most correct predictions against false positives. Then generate a new variable in your data set that classifies each student according to your chosen threshold.
 
-threshold.pred1 <- 
-
+threshold.pred1 <- 0.1
+for(i in (1:nrow(D1)))
```
Comment (charles-lang) [inline Assignment7.Rmd]: The ifelse() command is generally considered a faster implementation than a for loop. You may want to use that in future for simple variable renaming such as this.

File: Assignment7.Rmd [hunk 3]
```diff
@@ -41,30 +62,52 @@ plot(performance(pred.detail, "tpr", "fpr"))
 abline(0, 1, lty = 2)
 
 #Calculate the Area Under the Curve
-unlist(slot(performance(Pred2,"auc"), "y.values"))#Unlist liberates the AUC value from the "performance" object created by ROCR
+unlist(slot(performance(pred.detail,"auc"),"y.values"))
+#Unlist liberates the AUC value from the "performance" object created by ROCR
 
 #Now repeat this process, but using the variables you did not use for the previous model and compare the plots & results of your two models. Which one do you think was the better model? Why?
```
Comment (charles-lang) [inline Assignment7.Rmd]: Can you answer why you think the model you chose is superior?

File: Assignment7.Rmd [hunk 4]
```diff
@@ -41,30 +62,52 @@ plot(performance(pred.detail, "tpr", "fpr"))
...
+D1$recall.model1 <- (10+390)/(10+390+23+5)
 
 #Finally, calculate Kappa for your model according to:
 
 #First generate the table of comparisons
 table1 <- table(D1$level.up, D1$threshold.pred1)
-
+table1
 #Convert to matrix
 matrix1 <- as.matrix(table1)
```
Comment (charles-lang) [inline Assignment7.Rmd]: Did you generate diagnostics for two threshold values?

File: Assignment7.Rmd [hunk 5]
```diff
@@ -70,50 +71,49 @@ c.tree1_1 = rpart(level.up~pre.test.score+av.assignment.score+messages+forum.pos
...
-
-D1$precision.model1 <- (10+390)/(10+390)
-
-D1$recall.model1 <- (10+390)/(10+390+23+5)
-
+pred_table1 = table(D1$level.up, D1$threshold.pred1)
+pred_table1
+pred_table2 = table(D1$level.up, D1$threshold.pred2)
+pred_table2
+D1$accuracy.model1 <- (572+400)/(572+28+400)
```
Comment (charles-lang) [inline Assignment7.Rmd]: **Accuracy** should be the number of correct predictions divided by the total number of predictions.
**Precision** should be the number of students correctly classified as leveling up, divided by the sum of students correctly classified as leveling up plus the students incorrectly classified as leveling up
**Recall** should be the number of students correctly classified as leveling up, divided by the sum of students correctly classified as leveling up plus the students incorrectly classified as not leveling up when they did

Hope that helps!

Comment (edwardwangxy) [inline Assignment7.Rmd]: I think I did it correctly. So there will be only one accuracy, one precision and one recall for each.

Comment (charles-lang) [inline Assignment7.Rmd]: Yes, one accuracy, one precision and one recall for each threshold you test. You should test two thresholds.

**Pull request closed without merging**